exonum/src/events/noise/mod.rs

```diff
@@ -40,80 +42,144 @@ pub struct HandshakeParams {
     pub max_message_len: u32,
 }
 
+pub trait Handshake {
+    fn listen<S: AsyncRead + AsyncWrite + 'static>(self, stream: S) -> HandshakeResult<S>;
+
+    fn send<S: AsyncRead + AsyncWrite + 'static>(self, stream: S) -> HandshakeResult<S>;
+
+    fn read_handshake_msg<S: AsyncRead + 'static>(
```

Comment:
I'm against putting any methods other than `listen` and `send`, to the `Handshake` interface; gives too much details about the implementation.